### PR TITLE
feat(associate_roles): Add service and repository

### PR DESCRIPTION
**Pull request by VictorAvelar**
## Description

- Implement associate role repository
- Add associate role service
- Wire service and repository
- Test new service

## Context

Required for [Terraform associate role resource](https://github.com/labd/terraform-provider-commercetools/pull/414)

### CODE REVIEW

File: src/repositories/associate-role.ts [hunk 1]
```diff
@@ -1,14 +1,94 @@
-import type { AssociateRole } from '@commercetools/platform-sdk'
+import type { AssociateRole, AssociateRoleAddPermissionAction, AssociateRoleChangeBuyerAssignableAction, AssociateRoleDraft, AssociateRoleRemovePermissionAction, AssociateRoleSetCustomFieldAction, AssociateRoleSetNameAction, AssociateRoleSetPermissionsAction, Permission } from '@commercetools/platform-sdk'
```
Comment (demeyerthom) [inline src/repositories/associate-role.ts:1]: Can you run `pnpm run format` here? That way we keep a consistent code format

File: src/services/associate-roles.ts [hunk 2]
```diff
@@ -0,0 +1,18 @@
+import { Router } from "express";
+import { AbstractResourceRepository } from "../repositories/abstract";
```
Comment (demeyerthom) [inline src/services/associate-roles.ts:2]: We use explicit file extensions in this project, so could you add `.js` at the end of all the imports?

**Review by demeyerthom — CHANGES_REQUESTED** [review]
Hi @VictorAvelar looks good overal! Few minor comments i hope you can take a quick look at

**Comment by VictorAvelar** [comment]
@demeyerthom, I appreciate your feedback. The requested changes have been addressed. Thank you.

**Review by demeyerthom — APPROVED** [review]
LGTM

**Comment by demeyerthom** [comment]
@VictorAvelar merged. I will create a new release, this should be available in 30 minutes or so

**Pull request merged**